math/src/test/kotlin/ktx/math/rangesTest.kt

```diff
@@ -0,0 +1,244 @@
+    val divisor = 37
+    val divided = original / divisor
+    assertEquals(original.first / divisor, divided.first)
+    assertEquals(original.last / divisor, divided.last)
+  }
+
+  @Test
+  fun `should add to int range`() {
+    val original = -20..500
+    val addend = 37
+    val shifted = original + addend
+    assertEquals(original.first + addend, shifted.first)
+    assertEquals(original.last + addend, shifted.last)
+  }
+
+  @Test
+  fun `should subtract from int range`() {
+    val original = -20..500
+    val subtrahend = 37
+    val shifted = original - subtrahend
+    assertEquals(original.first - subtrahend, shifted.first)
+    assertEquals(original.last - subtrahend, shifted.last)
+  }
+
+  @Test
+  fun `should produce expected random int values`() {
+    val seed = 615843L
+    val start = 45
+    val endInclusive = 30654
+    val count = 10000
+    val directValues = mutableListOf<Int>()
+    val rangeValues = mutableListOf<Int>()
+    with(java.util.Random(seed)) {
+      repeat(count) {
+        directValues += nextInt(1 + endInclusive - start) + start
+      }
+    }
+    val range = start..endInclusive
+    with(java.util.Random(seed)) {
+      repeat(count) {
+        rangeValues += range.random(this)
+      }
+    }
+    assertEquals(directValues, rangeValues)
+  }
+
+  @Test
+  fun `should produce expected float range`() {
+    val center = -325f
+    val tolerance = 6000f
+    val range = center amid tolerance
+    assertEquals(range.start, center - tolerance)
+    assertEquals(range.endInclusive, center + tolerance)
+  }
+
+  @Test
+  fun `should multiply float range`() {
+    val original = -20f..500f
+    val multiplier = 37f
+    val scaled = original * multiplier
+    val invertedScaled = multiplier * original
+    assertEquals(original.start * multiplier, scaled.start)
+    assertEquals(original.endInclusive * multiplier, scaled.endInclusive)
+    assertEquals(original.start * multiplier, invertedScaled.start)
+    assertEquals(original.endInclusive * multiplier, invertedScaled.endInclusive)
+  }
+
+  @Test
+  fun `should divide float range`() {
+    val original = -20f..500f
+    val divisor = 37f
+    val divided = original / divisor
+    assertEquals(original.start / divisor, divided.start)
+    assertEquals(original.endInclusive / divisor, divided.endInclusive)
+  }
+
+  @Test
+  fun `should add to float range`() {
+    val original = -20f..500f
+    val addend = 37f
+    val shifted = original + addend
+    assertEquals(original.start + addend, shifted.start)
+    assertEquals(original.endInclusive + addend, shifted.endInclusive)
+  }
+
+  @Test
+  fun `should subtract from float range`() {
+    val original = -20f..500f
+    val subtrahend = 37f
+    val shifted = original - subtrahend
+    assertEquals(original.start - subtrahend, shifted.start)
+    assertEquals(original.endInclusive - subtrahend, shifted.endInclusive)
+  }
+
+  @Test
+  fun `should produce expected random float values`() {
```

Comment:
Same goes for this test. You can include an external math library in test scopes if that makes the asserts any easier.